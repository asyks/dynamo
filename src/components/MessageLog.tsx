import React from 'react'

import '../styles/App.css'

interface Props {
  messages: string[]
}

const MessageLog: React.FC<Props> = props => {
  const logElems: JSX.Element[] = []

  props.messages.forEach((value, i) => {
    logElems.push(
      <li key={i}>{value}</li>
    )
  })

  return <ul className="Web-socket-echo-log">{logElems}</ul>
}

export default MessageLog
