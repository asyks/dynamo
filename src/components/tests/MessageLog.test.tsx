import React from 'react'
import { render, unmountComponentAtNode } from 'react-dom'
import { act } from 'react-dom/test-utils'

import MessageLog from '../MessageLog'

let container: Element | null = null

beforeEach(() => {
  container = document.createElement("div")
  document.body.appendChild(container)
})

afterEach(() => {
  if (container !== null) {
    unmountComponentAtNode(container)
    container.remove()
    container = null
  }
})

it("renders with or without messages", () => {
  act(() => {
    render(<MessageLog messages={[]} />, container)
  })
  expect(container).toMatchSnapshot()

  act(() => {
    render(<MessageLog messages={["foo"]} />, container)
  })
  expect(container).toMatchSnapshot()

  act(() => {
    render(<MessageLog messages={["foo", "bar", "baz"]} />, container)
  })
  expect(container).toMatchSnapshot()

})
