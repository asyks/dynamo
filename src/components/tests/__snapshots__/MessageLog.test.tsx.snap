// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with or without messages 1`] = `
<div>
  <ul
    class="Web-socket-echo-log"
  />
</div>
`;

exports[`renders with or without messages 2`] = `
<div>
  <ul
    class="Web-socket-echo-log"
  >
    <li>
      foo
    </li>
  </ul>
</div>
`;

exports[`renders with or without messages 3`] = `
<div>
  <ul
    class="Web-socket-echo-log"
  >
    <li>
      foo
    </li>
    <li>
      bar
    </li>
    <li>
      baz
    </li>
  </ul>
</div>
`;
